corehq/apps/reports/static/reports/v2/js/context.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * todo add docstring
+ */
+
+hqDefine('reports/v2/js/context', [
+  'jquery',
+  'knockout',
+  'underscore',  // can we switch to lodash?
+  'hqwebapp/js/initial_page_data',
+], function (
+    $,
+    ko,
+    _,
+    initialPageData
+) {
+  'use strict';
+
+  var Endpoint = function (data) {
+    var self = this;
+
+    self.slug = data.slug;
+    self.urlname = data.urlname;
+
+    self.getUrl = function () {
+      return initialPageData
+              .reverse(self.urlname)
+              .replace('slug', self.slug);
```

Comment:
Instead of this, in the `registerurl` tag you can pass `'---'` instead of `'slug'` and then here you'd call `.reverse(self.urlname, self.slug)` and it'll do the substitution for you.

Reply:
fixed